server/src/pages/shot/editor.js

```diff
@@ -0,0 +1,150 @@
+
+exports.Editor = class Editor extends React.Component {
+  constructor(props) {
+    super(props);
+    this.draw = this.draw.bind(this);
+    this.setPosition = this.setPosition.bind(this);
+    this.state = {
+      tool: 'none',
+      color: '#000000',
+      size: '5'
+    };
+  }
+
+  render() {
+    let penState;
+    let highlighterState;
+    let canvasHeight = this.props.clip.image.dimensions.y;
+    let canvasWidth = this.props.clip.image.dimensions.x;
+    if (this.state.tool == 'highlighter') {
+      highlighterState = 'active';
+      penState = 'inactive';
+    } else if (this.state.tool == 'pen') {
+      highlighterState = 'inactive';
+      penState = 'active';
+    } else {
+      penState = 'inactive';
+      highlighterState = 'inactive';
+    }
+    return <div>
+      <div className="editor-header default-color-scheme">
+        <div className="shot-main-actions">
+          <a className={`button pen-button ${penState}`} id="pen" onClick={this.onClickPen.bind(this)} title="pen"></a>
+          <a className={`button highlight-button ${highlighterState}`} id="highlight" onClick={this.onClickHighlight.bind(this)} title="highlighter"></a>
+        </div>
+        <div className="shot-alt-actions">
+          <a className="button primary save" id="save" onClick={ this.onClickSave.bind(this) }>Save</a>
+          <a className="button secondary cancel" id="cancel" onClick={this.onClickCancel.bind(this)}>Cancel</a>
+        </div>
+      </div>
+      <div className="main-container">
+        <div className="canvas-container" id="canvas-container" ref={(canvasContainer) => this.canvasContainer = canvasContainer}>
+          <canvas className="image-holder centered" id="image-holder" ref={(image) => { this.imageCanvas = image }} height={ 2 * canvasHeight } width={ 2 * canvasWidth } style={{height: canvasHeight, width: canvasWidth}}></canvas>
+          <canvas className="highlighter centered" id="highlighter" ref={(highlighter) => { this.highlighter = highlighter }} height={canvasHeight} width={canvasWidth}></canvas>
+          <canvas className="editor centered" id="editor" ref={(editor) => { this.editor = editor }} height={canvasHeight} width={canvasWidth}></canvas>
+        </div>
+      </div>
+    </div>
+  }
+
+  componentDidUpdate() {
+    this.edit();
+  }
+
+  onClickCancel() {
+    this.props.onCancelEdit(false);
+    sendEvent("cancel", "annotation-toolbar");
+  }
+
+  onClickSave() {
+    sendEvent("save", "annotation-toolbar");
+    this.imageContext.drawImage(this.editor, 0, 0);
+    this.imageContext.globalCompositeOperation = 'multiply';
+    this.imageContext.drawImage(this.highlighter, 0, 0);
+    let dataUrl = this.imageCanvas.toDataURL();
+    this.props.onClickSave(dataUrl);
+  }
+
+  onClickHighlight() {
+    if (this.state.tool != 'highlighter') {
+      this.setState({tool: 'highlighter'});
+      sendEvent("highlighter-select", "annotation-toolbar");
+    } else {
+      this.setState({tool: 'none'});
```

Comment:
Does it make sense for there ever to be no tool? Seems like we're creating a state that is going to be confusing: the user wants to edit, they clicked edit, and now they are effectively just looking at an image and clicks do nothing.

If you can't have "no" tool active, then I'd imagine that clicking a button always activates that tool.